### ISSUE TITLE
The future of bestspeech for NVDA

### ISSUE THREAD

**Issue by M3198** [samtupy/b32tts_wrapper#4]
Hello,

Hope you are doing well.

What will be the future of bestspeech tts for NVDA, considering that NVDA will drop 32 bit support and will be 64 bit only? Will you guys be able to make a 64 bit addon for NVDA, or add some kind of emulation or communication bridge?
Best regards,
Muhammad.

**Comment by samtupy** [comment]
Hi,

There is no way the entire community will accept the loss of all 32 bit synthesizers, I just can't see that happening. If nobody else does it, I will write a bridge myself that lets 32 bit NVDA synth drivers communicate with the new 64 bit NVDA. I wouldn't be surprised if somebody else tackles this, but if nobody else wants the task, I'll gladly do it when I have time and energy.

**Comment by valiant8086** [comment]
Hi.

I'm pretty sure, not 100%, that NVDA's already got plans to get tts engines and other such add-ons to still work in 32bit. There's a bit of a discussion about that somewhere over there on their github.

**Comment by M3198** [comment]
Hello,

Hope you are doing well.
Is a 64 bit port being worked on? Just thought I'd drop in because we're now moving onto the beta phase for NVDA 2026. I feel work should be started from now so that by the time 2026 comes out we'll hopefully be ready.

**Comment by valiant8086** [comment]
there is intent of adding a more complete 32bit compatibility layer to NVDA, but it won't be in the 2026.1 release. They've got 32bit SAPI5 functional, and that is part of the plan, and they specifically mention 32bit synthesizer add-ons as a goal for this effort.

**Comment by M3198** [comment]
Hello,


Were you able to see my comment? Is work being done to port this addon to 64 bit?

Best regards,

Muhammad.

**Comment by valiant8086** [comment]
Full disclosure, I'm not a dev for this project.

I did read your original message. That's what I was responding to... But as far as I know they're using a dll which Rommix0 figured out how to fully interface with, without actually having the sourcecode to make the dll, and thus not a super easy path to recompile it in x64 or even arm64. They can hack it, by using a tiny x86 application to be a middle man. The x64 add-on will use this x86 application which is just a wrapper for the x86dll. It is working well for IBM tts, proving super efficient even on my terribly slow arm64 Galaxy Book Go.

So, if the dev wants to bother, this is probably the way it will go, but it is sort of a temporary need since NVDA itself is going to end up supporting the x86 version on its own sometime down the road.

**Comment by valiant8086** [comment]
> Hi,
> 
> There is no way the entire community will accept the loss of all 32 bit synthesizers, I just can't see that happening. If nobody else does it, I will write a bridge myself that lets 32 bit NVDA synth drivers communicate with the new 64 bit NVDA. I wouldn't be surprised if somebody else tackles this, but if nobody else wants the task, I'll gladly do it when I have time and energy.

here's the IBM TTS approach for this. I just recently read a message in a commit from Mic making it clear that they do intend to get NVDA to support 32bit synthesizer add-ons, so this is possibly a temporary need.

https://github.com/fastfinge/eloquence_64/

**Comment by M3198** [comment]
So can somebody please do something about this? Whenever I try to use Chris Duffley's addon which he made for 2026.1, I keep on getting synthesizer error could not load the bestspeech speech synthesizer. For reference, here's the addon.

[BeSTspeech-64-bit-NVDA.zip](https://github.com/user-attachments/files/28099134/BeSTspeech-64-bit-NVDA.zip)

**Comment by jcoffin1** [comment]
Hi. 

I'm waiting for my pull request to be reviewed, but with the help of Codex I fixed the compatibility issues with this and it's now working on NVDA 2026.2. 

I have forked the project to my own github account. I'm not sure what to do about uploading it because it's not my add-on, but if someone can give me some guidance  I'd be happy to upload it and you guys can use it.
